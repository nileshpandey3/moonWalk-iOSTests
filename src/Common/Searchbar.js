import React, { Component } from "react";
import styled from "styled-components";
import Icon from "react-native-vector-icons/FontAwesome5";

const Wrapper = styled.View`
  width: 80%;
  padding: 15px 20px;
  border-radius: 40px;
  font-size: 20px;
  margin: 20px;
  color: white;
  background: white;
  flex-direction: row;
  align-items: center;
`;

const StyledInput = styled.TextInput`
  flex: 1;
  margin-left: 10px;
`;

class SearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchStr: ""
    };
  }

  render() {
    const { searchStr } = this.state;
    return (
      <Wrapper>
        <Icon name="search" size={20} color="#42446f" />
        <StyledInput
          value={searchStr}
          placeholder="Search launches..."
          placeholderTextColor="#777"
          returnKeyType="search"
          clearButtonMode="always"
          onChangeText={searchStr => this.setState({ searchStr })}
          onSubmitEditing={() => this.props.launchSearch(searchStr)}
        />
      </Wrapper>
    );
  }
}

export default SearchBar;
