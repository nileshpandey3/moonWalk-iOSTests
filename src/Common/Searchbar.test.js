import React from "react";
import Searchbar from "./Searchbar";
import renderer from "react-test-renderer";

describe("Searchbar", () => {
  it("renders correctly", () => {
    const props = {
      launchSearch: jest.fn()
    };
    const tree = renderer.create(<Searchbar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
