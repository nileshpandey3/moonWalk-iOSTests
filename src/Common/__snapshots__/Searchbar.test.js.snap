// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 40,
        "color": "white",
        "flexDirection": "row",
        "fontSize": 20,
        "marginBottom": 20,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 20,
        "paddingBottom": 15,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 15,
        "width": "80%",
      },
      undefined,
    ]
  }
>
  <Icon
    color="#42446f"
    name="search"
    size={20}
  />
  <TextInput
    allowFontScaling={true}
    clearButtonMode="always"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Search launches..."
    placeholderTextColor="#777"
    rejectResponderTermination={true}
    returnKeyType="search"
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginLeft": 10,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
